src/main/java/build/buildfarm/worker/shard/ShardWorkerContext.java

```diff
@@ -488,6 +499,16 @@ private void insertBlob(Digest digest, ByteString content) throws InterruptedExc
   }
 
   private void insertFile(Digest digest, Path file) throws IOException, InterruptedException {
+
+    if (keepInsertFiles) {
+      insertFileToLocalStorage(digest, file);
+    } else {
+      insertFileToCasMember(digest, file);
```

Comment:
Generalize these to consume a single Write

Reply:
That will also handle the only other comment below, which will ensure that try-with-resources is used on the FeedbackOutputStream